src/components/drawers/EditKeyResult.vue

```diff
@@ -356,6 +374,90 @@ export default {
       }
     },
 
+    async getObjectiveContributors() {
+      const objectiveRef = await db.doc(`objectives/${this.objective.id}`);
+      const objectiveContributors = await db
+        .collection('objectiveContributors')
+        .where('objective', '==', objectiveRef)
+        .where('archived', '==', false)
+        .get()
+        .then((snapshot) => snapshot.docs)
+        .then((docs) => docs.map((d) => d.data()));
+
+      const contributors = await Promise.all(
+        objectiveContributors.map(async (con) => {
+          return {
+            ref: await con.item.get(),
+            name: await con.item.get().then((snapshot) => snapshot.data().name),
+          };
+        })
+      );
+      return contributors;
+    },
+
+    async getKeyResultOwners() {
+      const objectiveRef = await db.doc(`objectives/${this.objective.id}`);
+      const keyResults = await db
+        .collection('keyResults')
+        .where('objective', '==', objectiveRef)
+        .where('archived', '==', false)
+        .get()
+        .then((snapshot) => snapshot.docs)
+        .then((docs) => docs.map((d) => d.data()));
+
+      const keyResultNames = await Promise.all(
+        keyResults.map(async (owner) => {
+          return {
+            ref: await owner.parent.get(),
+            name: await owner.parent.get().then((snapshot) => snapshot.data().name),
+          };
+        })
+      );
+      return keyResultNames;
+    },
+
+    async syncObjectiveContributor() {
+      const keyResultOwners = await this.getKeyResultOwners();
+      const contributors = await this.getObjectiveContributors();
+
+      let contributorsCopy = [...contributors];
+      let keyResultOwnersCopy = [...keyResultOwners];
+
+      // Filter out already present links
+      contributors.map(async (c) => {
+        keyResultOwners.map(async (k) => {
```

Comment:
Disse trenger ikke å være `async`? Og bør bruke `forEach` i stedet for `map` siden vi ikke er interessert i noen returverdier.